app/assets/src/components/views/amr_heatmap/AMRHeatmapVis.jsx

```diff
@@ -104,10 +103,28 @@ export default class AMRHeatmapVis extends React.Component {
     return heatmapValues;
   }
 
-  renderHeatmap() {
+  assembleHeatmapData() {
     const rows = this.getHeatmapLabels();
     const columns = this.state.sampleLabels;
     const values = this.computeHeatmapValues(rows);
+    return [rows, columns, values];
+  }
+
+  createHeatmap() {
```

Comment:
`createHeatmap` and `updateHeatmap` seem very similar (they both call `assembleHeatmapData`, then call functions that take in the same arguments) — I'd maybe combine the logic of whether to update vs render + corresponding update/rendering call in `renderHeatMap` and have all that in one method, so when reading the code there's less code-tracing going around between similar methods (could also just be a preferential thing though)